spv/sync.go

```diff
@@ -385,6 +392,12 @@ func (s *Syncer) Run(ctx context.Context) error {
 		}
 		s.fetchHeadersFinished()
 
+		if s.birthday != nil && !s.birthdaySet.Swap(true) {
```

Comment:
Since you don't store `birthdaySet`, you rebuild the birthday every time the wallet restarts, which is not great from a performance pov.

Reply:
Now storing this info.